lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -77,6 +77,39 @@ class DecomposeDotOperand : public mlir::RewritePattern {
       rewriter.replaceOpWithNewOp<triton::gpu::ConvertLayoutOp>(op, dstType,
                                                                 tmp);
       return mlir::success();
+    } else if (srcType.getEncoding().isa<triton::gpu::BlockedEncodingAttr>() &&
+               dstType.getEncoding().isa<triton::gpu::SharedEncodingAttr>()) {
+      auto result = convert.getResult();
+      if (!result.hasOneUse())
+        return mlir::failure();
+
+      if (!llvm::isa<triton::DotOp>(*(result.user_begin())))
+        return mlir::failure();
+
+      if (result.use_begin()->getOperandNumber() != 0)
+        return mlir::failure();
+
+      SetVector<Operation *> bwdSlices;
+      mlir::getBackwardSlice(result, &bwdSlices);
+      auto firstDotOp = llvm::find_if(
+          bwdSlices, [](Operation *op) { return isa<triton::DotOp>(op); });
```

Comment:
Can we guarantee the decomposed convertLayoutOp can always be optimized away by this way?
I'm considerting if we need to check the op types along the path

Reply:
yes, for currently most common work loads.